sympy/sets/conditionset.py

```diff
@@ -34,6 +34,8 @@ def __new__(cls, sym, condition, base_set):
             return S.EmptySet
         if condition == S.true:
             return base_set
+        if getattr(base_set, "is_FiniteSet", False) and base_set.is_FiniteSet:
+            return base_set.intersect(condition.as_set())
```

Comment:
It doesn't -- it just checks to see whether the condition evaluates to True, False or other (which becomes None with fuzzy_bool).